File: flutter_customer/lib/models/profile_model.dart

class ProfileModel {
  final int customerId;
  final String firstName;
  final String lastName;
  final String? email;
  final String? phone;
  final String? profileImage;
  final String? gender;
  final String? city;
  final String? state;
  final String? dateOfBirth;
  final bool isVerified;

  // Style profile
  final int? styleProfileId;
  final String? apparelSize;
  final String? fitPreference;
  final List<String> preferredColors;
  final List<String> preferredStyles;

  ProfileModel({
    required this.customerId,
    required this.firstName,
    required this.lastName,
    this.email,
    this.phone,
    this.profileImage,
    this.gender,
    this.city,
    this.state,
    this.dateOfBirth,
    required this.isVerified,
    this.styleProfileId,
    this.apparelSize,
    this.fitPreference,
    this.preferredColors = const [],
    this.preferredStyles = const [],
  });

  String get fullName {
    return '$firstName $lastName'.trim();
  }

  factory ProfileModel.fromJson(Map<String, dynamic> json) {
    return ProfileModel(
      customerId:
          int.tryParse(json['customer_id'].toString()) ?? 0,

      firstName:
          json['first_name']?.toString() ?? '',

      lastName:
          json['last_name']?.toString() ?? '',

      email:
          json['email']?.toString(),

      phone:
          json['phone']?.toString(),

      profileImage:
          json['profile_image']?.toString(),

      gender:
          json['gender']?.toString(),

      city:
          json['city']?.toString(),

      state:
          json['state']?.toString(),

      dateOfBirth:
          json['date_of_birth']?.toString(),

      isVerified:
          json['is_verified'] == true,

      styleProfileId:
          json['style_profile_id'] != null
              ? int.tryParse(
                  json['style_profile_id'].toString(),
                )
              : null,

      apparelSize:
          json['apparel_size']?.toString(),

      fitPreference:
          json['fit_preference']?.toString(),

      preferredColors:
          _stringList(json['preferred_colors']),

      preferredStyles:
          _stringList(json['preferred_styles']),
    );
  }

  static List<String> _stringList(dynamic value) {
    if (value == null) {
      return [];
    }

    if (value is List) {
      return value
          .map((item) => item.toString())
          .toList();
    }

    return [];
  }
}

File: backend/models/customer/profile.model.js
const pool = require("../../config/db");

/**
 * Get the logged-in customer's complete profile.
 *
 * Combines:
 *   customers
 *   customer_style_profiles
 */
exports.getProfile = async (customerId) => {
    const result = await pool.query(
        `
        SELECT
            c.customer_id,
            c.first_name,
            c.last_name,
            c.email,
            c.phone,
            c.profile_image,
            c.gender,
            c.city,
            c.state,
            TO_CHAR(c.date_of_birth, 'YYYY-MM-DD') AS date_of_birth, 
            c.is_verified,
            c.created_at,
            c.updated_at,

            sp.profile_id AS style_profile_id,
            sp.apparel_size,
            sp.fit_preference,
            sp.preferred_colors,
            sp.preferred_styles

        FROM customers c

        LEFT JOIN customer_style_profiles sp
            ON sp.customer_id = c.customer_id

        WHERE c.customer_id = $1

        LIMIT 1
        `,
        [customerId]
    );

    return result.rows[0] || null;
};

exports.updateProfile = async (
    customerId,
    {
        firstName,
        lastName,
        gender,
        dateOfBirth,
        phone,
        email,
        city,
        state,
    }
) => {

    const result = await pool.query(
        `
        UPDATE customers
        SET
            first_name = COALESCE($2, first_name),
            last_name = COALESCE($3, last_name),
            gender = COALESCE($4, gender),
            date_of_birth = COALESCE($5, date_of_birth),

            email = COALESCE($6, email),
            city = COALESCE($7, city),
            state = COALESCE($8, state),

            updated_at = CURRENT_TIMESTAMP

        WHERE customer_id = $1

        RETURNING
            customer_id,
            first_name,
            last_name,
            email,
            phone,
            profile_image,
            gender,
            city,
            state,
            TO_CHAR(date_of_birth, 'YYYY-MM-DD') AS date_of_birth,
            is_verified,
            created_at,
            updated_at
        `,
        [
            customerId, // $1
            firstName,  // $2
            lastName,   // $3
            gender,     // $4
            dateOfBirth,// $5
            email,      // $6
            city,       // $7
            state,      // $8
        ]
    );

    return result.rows[0] || null;
};